### DISCUSSION TITLE
How to download file from external URL

### DISCUSSION THREAD

**Discussion by D-Marc1** [ionic-team/capacitor#3557] [Q&A]
How would I download a file from an external URL and save it to the file system in Capacitor?

**Comment by filipegeric** [comment]
You could try using this [plugin](https://ionicframework.com/docs/native/file-transfer). It should be compatible with Capacitor. If it doesn't suit your needs you could write your own plugin ([https://capacitorjs.com/docs/plugins](https://capacitorjs.com/docs/plugins)) for downloading file in the background. 
You could also download a file using JavaScript and then store it on the device using [Filesystem plugin](https://capacitorjs.com/docs/apis/filesystem), but then you would have to keep your app open while the file is downloading.

**Reply by D-Marc1** [reply]
My issue was that it wasn't working with presigned URLs, like in the browser on desktop. I returned a base64 instead, then I downloaded that to the device with the the Filesystem plugin and that worked perfectly
